tests/trajectory.cpp

```diff
@@ -167,13 +167,37 @@ TEST_CASE("Specify a format parameter") {
     Trajectory file("data/xyz/helium.xyz.but.not.really", 'r', "XYZ");
     auto frame = file.read();
     CHECK(frame.size() == 125);
+
+    auto tmpfile = NamedTempPath(".xyz");
+    file = Trajectory(tmpfile, 'w', "XYZ / GZ");
+    frame = Frame();
+    frame.add_atom(Atom("Fe"), {0, 1, 2});
+    file.write(frame);
+    file.close();
+
+    // Full format specification
+    file = Trajectory(tmpfile, 'r', "XYZ / GZ");
+    frame = file.read();
+    CHECK(frame.size() == 1);
+    CHECK(frame[0].name() == "Fe");
+
+    // only the compression method, the format will be guessed from extension
```

Comment:
Should there be a test to see if both format and extension can be guessed?

Reply:
What do you mean ? `Trajectory("foo.xzy.gz", 'r', "")`?

Reply:
Yes

Reply:
This is tested in https://github.com/chemfiles/chemfiles/blob/377b72eb9aca4893c2f9e72a67ec7be03e84da63/tests/files/xz-file.cpp#L10-L15

Reply:
This is tested in https://github.com/chemfiles/chemfiles/blob/377b72eb9aca4893c2f9e72a67ec7be03e84da63/tests/formats/compressed-xyz.cpp#L43-L53

Reply:
Must have missed that

Reply:
This was added in a separated PR (I think #166)